File: doc-examples/nitrite-example-java/src/main/java/example/PersonWithDateOfBirthRepository.java
package example;

import io.micronaut.data.nitrite.annotation.NitriteRepository;
import io.micronaut.data.repository.CrudRepository;
import io.micronaut.data.repository.jpa.JpaSpecificationExecutor;

// tag::personWithDateOfBirthRepository[]
@NitriteRepository
public interface PersonWithDateOfBirthRepository extends CrudRepository<PersonWithDateOfBirth, String>, JpaSpecificationExecutor<PersonWithDateOfBirth> {

    // tag::personWithDateOfBirthRepository-countByName[]
    long countByName(String name);
    // end::personWithDateOfBirthRepository-countByName[]
}
// end::personWithDateOfBirthRepository[]

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/PersonWithDateOfBirthRepository.kt
package example

import io.micronaut.data.nitrite.annotation.NitriteRepository
import io.micronaut.data.repository.CrudRepository
import io.micronaut.data.repository.jpa.JpaSpecificationExecutor

// tag::personWithDateOfBirthRepository[]
@NitriteRepository
interface PersonWithDateOfBirthRepository : CrudRepository<PersonWithDateOfBirth, String>, JpaSpecificationExecutor<PersonWithDateOfBirth> {

    // tag::personWithDateOfBirthRepository-countByName[]
    fun countByName(name: String): Long
    // end::personWithDateOfBirthRepository-countByName[]
}
// end::personWithDateOfBirthRepository[]
